src/libraries/System.DirectoryServices.Protocols/tests/AsqRequestControlTests.cs

```diff
@@ -17,13 +19,29 @@ public void Ctor_Default()
             Assert.True(control.ServerSide);
             Assert.Equal("1.2.840.113556.1.4.1504", control.Type);
 
-            Assert.Equal(new byte[] { 48, 132, 0, 0, 0, 2, 4, 0 }, control.GetValue());
+            var expected = (Environment.OSVersion.Platform == PlatformID.Win32NT) ? new byte[] { 48, 132, 0, 0, 0, 2, 4, 0 } : new byte[] { 48, 2, 4, 0 };
+
+            Assert.Equal(expected, control.GetValue());
+        }
+
+        public static IEnumerable<object[]> Ctor_String_Test_data()
+        {
+            if (Environment.OSVersion.Platform == PlatformID.Win32NT)
```

Comment:
```suggestion
            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Reply:
Good catch, we actually have helper methods we are supposed to use in tests, PlatformDetection and I forgot to use it here